config/routes.rb

```diff
@@ -4,6 +4,7 @@
   mount_devise_token_auth_for 'User', at: 'auth'
   get 'books', to: 'books#index'
   get 'books/shows' , to: 'books#shows'
+  get 'books/test', to: 'books#test'
```

Comment:
If you apply the previous suggestion of `resources`, this could be refactored to something cleaner. This is an exaple with cars
```ruby
resources :cars, only: [:create, :show, :index] do
  collection do
    get :find_by_model
  end
end
```